docs/en/sampling_params.md

```diff
@@ -52,6 +55,26 @@ spaces_between_special_tokens: bool = True,
 regex: Optional[str] = None,
 # Do parallel sampling and return `n` outputs.
 n: int = 1,
+
+## Penalties. See [Performance Implications on Penalties] section below for more informations.
+
+# Float that penalizes new tokens based on their frequency in the generated text so far.
+# Values > 0 encourage the model to use new tokens, while values < 0 encourage the model to
+# repeat tokens.
+frequency_penalty: float = 0.0,
+# Float that penalizes new tokens based on whether they appear in the generated text so far.
+# Values > 0 encourage the model to use new tokens, while values < 0 encourage the model to repeat
+# tokens.
+presence_penalty: float = 0.0,
+# Float that penalizes new tokens based on whether they appear in the prompt and the generated text
+# so far. Values > 1 encourage the model to use new tokens, while values < 1 encourage the model to
+# repeat tokens.
+repetition_penalty: float = 1.0,
```

Comment:
If there's already `frequency_penalty` and `presence_penalty`, is there still a need for `repetition_penalty`? Can the first two not address certain situations?

Reply:
When I was implementing the penalty logic, I wrote based on [what it is in vLLM docs](https://github.com/vllm-project/vllm/blob/main/vllm/sampling_params.py#L49-L60), and [how they implemented](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/sampler.py#L201-L268).

They have `frequency_penalty`, `presence_penalty`, and `repetition_penalty` all in place.

`frequency_penalty` is to penalize based on frequency in the output tokens.
`presence_penalty` is to penalize based on the presence (1 or 0) in the output token.

`repetition_penalty` is based on the presence (1 or 0) in the input + output tokens. So it is slightly different than `presence_penalty`.

I would personally want to implement additional penalty for frequency in the input + output token as well. While it is easy to do, I had hard time naming the penalty, and also vLLM does not have this, so I just skipped it for now.

Reply:
ok